### PR TITLE
22 move changes from crypto3 math up to commit 2c3d8bb40f595fc7b87c8d46f7466c09dbc3731b

### LINKED ISSUE #22: Move changes from crypto3-math up to commit 2c3d8bb40f595fc7b87c8d46f7466c09dbc3731b

### PR DESCRIPTION
**Pull request by martun**
Changes are related to caching the omega values in evaluation domains.

### CODE REVIEW

File: test/basic_radix2_domain.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+#include <nil/actor/math/domains/detail/basic_radix2_domain_aux.hpp>
+
+#include <nil/crypto3/algebra/random_element.hpp>
+
+using namespace nil::crypto3::algebra;
+using namespace nil::actor::math;
+
+typedef fields::bls12_fr<381> FieldType;
+
+/* This test is disabled in crypto3, it's actually a benchmark, runs slow.
```
Comment (Zerg1996) [inline test/basic_radix2_domain.cpp:47]: Let's crate mb directory with benchs? I thinks we need to set correct flags and optimisation for such type of code.

Comment (Zerg1996) [reply test/basic_radix2_domain.cpp:47]: (maybe it's additional issue task)

File: test/evaluation_domain.cpp [hunk 2]
```diff
@@ -1,7 +1,6 @@
 //---------------------------------------------------------------------------//
 // Copyright (c) 2020-2021 Mikhail Komarov <nemo@nil.foundation>
 // Copyright (c) 2020-2021 Nikita Kaskov <nbering@nil.foundation>
-// Copyright (c) 2022 Aleksei Moskvin <alalmoskvin@nil.foundation>
```
Comment (Zerg1996) [inline test/evaluation_domain.cpp:4]: Why :D

**Review by Zerg1996 — APPROVED** [review]

File: test/basic_radix2_domain.cpp (hunk 1 above)
Comment (martun) [reply test/basic_radix2_domain.cpp:47]: Yes, we will have one on the new actor code. I was asking this questions in some channel, to decide on it.

File: test/evaluation_domain.cpp (hunk 2 above)
Comment (martun) [reply test/evaluation_domain.cpp:4]: Yep, git patch remove it, probably it did not find it in the crypto3-math :)

Comment (Zerg1996) [reply test/evaluation_domain.cpp:4]: Oh, I see. That's true - I didn't change this file in crypto3, only in actor version.

File: test/basic_radix2_domain.cpp (hunk 1 above)
Comment (Zerg1996) [reply test/basic_radix2_domain.cpp:47]: If u want - you can also adding me into conversation!

**Pull request merged**